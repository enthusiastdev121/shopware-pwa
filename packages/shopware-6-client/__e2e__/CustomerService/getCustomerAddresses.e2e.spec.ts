import {
  update,
  login,
  getCustomerAddresses,
} from "@shopware-pwa/shopware-6-client";

describe("shopware-6-client - E2E - CustomerService - getCustomerAddresses", () => {
  beforeEach(() => {
    update({ contextToken: "" });
  });

  it("should fetch customer addresses but test only one from array", async () => {
    await login({ username: "test.e2e@test.pl", password: "password" });
    const result = await getCustomerAddresses();
    const fakedResult = result
      .map((address: any) => {
        address._uniqueIdentifier = "mockedUniqueIdentifier";
        address.city = "mockedCity";
        address.createdAt = "mockedCreatedAt";
        address.zipcode = "mockedZipcode";
        address.street = "mockedStreet";
        address.firstName = "mockedFirstName";
        address.lastName = "mockedLastName";
        address.id = "mockedId";
        return address;
      })
      .shift();
    expect(fakedResult).toMatchSnapshot();
  });

  it("should return error when fetching customer addresses without logged user", async () => {
    try {
      await getCustomerAddresses();
      expect("didn't throw an error").toEqual("should throw an error");
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });
});
