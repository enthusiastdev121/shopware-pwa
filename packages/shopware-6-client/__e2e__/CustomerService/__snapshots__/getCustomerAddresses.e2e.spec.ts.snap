// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - CustomerService - getCustomerAddresses should fetch customer addresses 1`] = `
Array [
  Object {
    "_uniqueIdentifier": "c03217761a5847c19ea3d9801dd2bb7d",
    "additionalAddressLine1": null,
    "additionalAddressLine2": null,
    "city": "TestTown",
    "company": null,
    "country": null,
    "countryId": "38245a84c3d5425b8bac97fc845b5ddd",
    "countryState": null,
    "countryStateId": null,
    "createdAt": "2020-01-01T20:14:14+00:00",
    "customFields": null,
    "customer": null,
    "customerId": "d5ba879ae22f4dcd8f881046dbcedc41",
    "department": null,
    "extensions": Object {
      "foreignKeys": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
    },
    "firstName": "test",
    "id": "c03217761a5847c19ea3d9801dd2bb7d",
    "lastName": "e2e",
    "phoneNumber": null,
    "salutation": null,
    "salutationId": "c370eb5cd1df4d4dbcc78f055b693e79",
    "street": "teststreet",
    "title": null,
    "translated": Array [],
    "updatedAt": null,
    "vatId": null,
    "versionId": null,
    "zipcode": "10-900",
  },
]
`;

exports[`shopware-6-client - E2E - CustomerService - getCustomerAddresses should return error when fetching customer addresses without logged user 1`] = `[Error: Request failed with status code 403]`;
